### PR TITLE
Possibility to add custom serializers

### PR DESCRIPTION
**Pull request by wellguimaraes**
Now it's possible to add custom serializers.

### CODE REVIEW

File: paperdb/build.gradle [hunk 1]
```diff
@@ -31,6 +31,7 @@ android {
 dependencies {
     compile 'com.esotericsoftware:kryo:3.0.1'
     compile 'de.javakaffee:kryo-serializers:0.30'
+    compile 'joda-time:joda-time:2.8.1'
```
Comment (pilgr) [inline paperdb/build.gradle]: This dependency should be optional

File: paperdb/src/main/java/io/paperdb/DbStoragePlainFile.java [hunk 2]
```diff
@@ -63,6 +69,10 @@ private Kryo createKryoInstance() {
                 new NoArgCollectionSerializer());
         // To keep backward compatibility don't change the order of serializers above
 
+		// UUID and joda DateTime support
+		kryo.register(UUID.class, new UUIDSerializer());
+		kryo.register(DateTime.class, new JodaDateTimeSerializer());
```
Comment (pilgr) [inline paperdb/src/main/java/io/paperdb/DbStoragePlainFile.java]: Register joda serializer only if joda lib is included in app's dependencies.

Comment (wellguimaraes) [inline paperdb/src/main/java/io/paperdb/DbStoragePlainFile.java]: What about registering custom serializers using something like `<T> void Paper.registerSerializer(Class<T> type, Serializer<T> typeSerializer)`?

Comment (pilgr) [inline paperdb/src/main/java/io/paperdb/DbStoragePlainFile.java]: Right, would be much better to have such extension point.

**Comment by pilgr** [comment]
We push all api changes in development branch. Could you please rebase your patch on top of development branch because it has a bit different architecture to manage storage instances.

**Pull request closed without merging**